var/spack/repos/builtin/packages/vtk/package.py

```diff
@@ -195,34 +204,36 @@ def cmake_args(self):
             if '+mpi' in spec:
                 cmake_args.extend(["-DModule_vtkIOParallelXdmf3:BOOL=ON"])
 
-        if '+osmesa' in spec:
-            prefix = spec['mesa'].prefix
-            osmesa_include_dir = prefix.include
-            osmesa_library = os.path.join(prefix.lib, 'libOSMesa.so')
+        cmake_args.extend([
+            '-DVTK_USE_SYSTEM_GLEW:BOOL=ON',
 
-            use_param = 'VTK_USE_X'
-            if 'darwin' in spec.architecture:
-                use_param = 'VTK_USE_COCOA'
+            '-DVTK_RENDERING_BACKEND:STRING=OpenGL{0}'.format(
+                '2' if '+opengl2' in spec else ''),
+        ])
 
+        prefix = spec['gl'].prefix
+        has_osmesa = ('+osmesa' in spec)
```

Comment:
This is only in one place now so no need to factor it out into a seperate variable, it can be collapsed into just `if '+osmesa' in spec:`